### PR TITLE
add more tests for pub, api classes

### PR DESCRIPTION
**Pull request by devoncarew**
Add more tests for pub and for the api classes.

@lukechurch

### CODE REVIEW

File: lib/src/common.dart [hunk 1]
```diff
@@ -63,3 +63,15 @@ class Lines {
     return _starts.length;
   }
 }
+
+/// If [str] has leading and trailing quotes, remove them.
+String stripQuotes(String str) {
```
Comment (lukechurch) [inline lib/src/common.dart]: Rename to 'stripMatchingQuotes'?

File: lib/src/pub.dart [hunk 2]
```diff
@@ -338,16 +339,7 @@ Set<String> getAllUnsafeImportsFor(String dartSource) {
...
-        if (str.startsWith("'") && str.endsWith("'")) {
-          if (str.length > 1) {
-            str = str.substring(1, str.length - 1);
-          }
-        } else if (str.startsWith('"') && str.endsWith('"')) {
-          if (str.length > 1) {
-            str = str.substring(1, str.length - 1);
-          }
-        }
+        String str = stripQuotes(token.lexeme);
```
Comment (lukechurch) [inline lib/src/pub.dart]: +1 :)

**Comment by lukechurch** [comment]
One naming nit, then lgtm. Thanks!

**Comment by devoncarew** [comment]
Renamed the method. Also, added a few more tests for uncovered code; sorry, I should have done that in a separate PR.

**Comment by lukechurch** [comment]
No worries, I'll take another look

**Comment by lukechurch** [comment]
merging

**Pull request merged**